MainRepository/Program.cs

```diff
@@ -1,25 +1,26 @@
 ﻿// Program to print a diamond using an asterisk
 Console.Write ("Enter the no of rows: ");
-int n = Convert.ToInt32 (Console.ReadLine ());
-for (int i = 0; i < n; i++) {
-   for (int j = 0; j < n; j++) {
-      if (i + j >= n - 1) Console.Write ("* ");
-      else Console.Write ("  ");
+if (int.TryParse (Console.ReadLine (), out int n) && n > 0) {
+   for (int i = 0; i < n; i++) {
+      for (int j = 0; j < n; j++) {
+         if (i + j >= n - 1) Console.Write ("* ");
+         else Console.Write ("  ");
+      }
+      for (int j = 0; j < n; j++) {
+         if (i > j) Console.Write ("* ");
+         else Console.Write ("  ");
+      }
+      Console.WriteLine ();
    }
-   for (int j = 0; j < n; j++) {
-      if (i > j) Console.Write ("* ");
-      else Console.Write ("  ");
+   for (int i = 1; i < n; i++) {
+      for (int j = 0; j < n; j++) {
+         if (i <= j) Console.Write ("* ");
+         else Console.Write ("  ");
+      }
+      for (int j = 0; j < n; j++) {
+         if (i + j < n - 1) Console.Write ("* ");
+         else Console.Write ("  ");
+      }
+      Console.WriteLine ();
    }
-   Console.WriteLine ();
-}
-for (int i = 1; i < n; i++) {
-   for (int j = 0; j < n; j++) {
-      if (i <= j) Console.Write ("* ");
-      else Console.Write ("  ");
-   }
-   for (int j = 0; j < n; j++) {
-      if (i + j < n - 1) Console.Write ("* ");
-      else Console.Write ("  ");
-   }
-   Console.WriteLine ();
-}
\ No newline at end of file
+} else Console.WriteLine ("Enter the valid input");
```

Comment:
In Metamation, we usually write like this:
if (!int.TryParse (Console.ReadLine (), out int n)) return;